e_methods.rb

```diff
@@ -0,0 +1,155 @@
+      my_each { |key, value| result[key] = value if yield(key, value) }
+    end
+    result
+  end
+
+  def my_all?(argum = nil)
+    if block_given? && !argum.nil?
+      puts 'warning: given block not used'
+      my_each { |element| return false unless element.is_a? argum }
+    elsif block_given?
+      my_each { |element| return false unless yield(element) }
+    elsif argum.nil?
+      my_each { |element| return false if element.nil? || element === false }
+    elsif !argum.nil? && (argum.is_a? Class)
+      my_each { |element| return false unless element.is_a? argum }
+    elsif !argum.nil? && argum.instance_of?(Regexp)
+      my_each { |element| return false unless argum.match(element) }
+    else
+      my_each { |element| return false unless element }
+    end
+    true
+  end
+
+  def my_any?(argum = nil)
+    if block_given? && !argum.nil?
+      puts 'warning: given block not used'
+      my_each { |element| return true if element.is_a? argum }
+    elsif block_given?
+      my_each { |element| return true if yield(element) == true }
+    elsif argum.nil?
+      my_each { |element| return true if element.nil? || element == false }
+    elsif !argum.nil? && (argum.is_a? Class)
+      my_each { |element| return true if element.is_a? argum }
+    elsif !argum.nil? && argum.instance_of?(Regexp)
+      my_each { |element| return true if argum.match(element) }
+    elsif argum
+      my_each { |element| return true if element == argum }
+    else
+      my_each { |element| return true if element }
+    end
+    false
+  end
+
+  def my_none?(argum = nil)
+    if block_given? && !argum.nil?
+      puts 'warning: given block not used'
+      my_each { |element| return true unless element.is_a? argum }
+    elsif block_given?
+      my_each { |element| return true unless yield(element) == true }
+    elsif argum.nil?
+      my_each { |element| return true if element.nil? || element == false }
+    elsif !argum.nil? && (argum.is_a? Class) # >-
+      my_each { |element| return true unless element.is_a? argum }
+    elsif !argum.nil? && argum.instance_of?(Regexp)
+      my_each { |element| return true unless argum.match(element) }
+    elsif argum
+      my_each { |element| return true unless element == argum }
+    else
+      my_each { |element| return true unless element }
+    end
+    false
+  end
+
+  def my_count(argum = nil)
+    counter = 0
+    if block_given?
+      my_each { |element| counter += 1 if yield(element) == true }
+    elsif argum
+      my_each { |element| counter += 1 if element == argum }
+    else
+      my_each { |element| counter += 1 if element }
+    end
+    counter
+  end
+
+  def my_map(prc = nil)
+    return to_enum(:my_map) unless block_given? || !prc.nil?
+
+    new_arr = []
+    if !prc.nil?
+      to_a.my_each { |i| new_arr.push(prc.call(i)) }
+    else
+      to_a.my_each { |i| new_arr.push(yield(i)) }
+    end
+    new_arr
+  end
+
+  def my_inject(argum = nil, sym = nil)
+    if !block_given?
```

Comment:
Should raise a "LocalJumpError" when no block or argument is given
     Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)
       expected LocalJumpError but nothing was raised